skbio/stats/ordination/_principal_coordinate_analysis.py

```diff
@@ -62,37 +82,52 @@ def pcoa(distance_matrix):
 
     Notes
     -----
-    It is sometimes known as metric multidimensional scaling or
-    classical scaling.
-
-    .. note::
-
-       If the distance is not euclidean (for example if it is a
-       semimetric and the triangle inequality doesn't hold),
-       negative eigenvalues can appear. There are different ways
-       to deal with that problem (see Legendre & Legendre 1998, \S
-       9.2.3), but none are currently implemented here.
-
-       However, a warning is raised whenever negative eigenvalues
-       appear, allowing the user to decide if they can be safely
-       ignored.
+    Principal Coordinate Analysis is sometimes known as
+    metric multidimensional scaling or classical scaling.
+
+    If the distance is not euclidean (for example if it is a
+    semimetric and the triangle inequality doesn't hold),
+    negative eigenvalues can appear. There are different ways
+    to deal with that problem (see Legendre & Legendre 1998, \S
+    9.2.3), but none are currently implemented here.
+
+    However, a warning is raised whenever negative eigenvalues
+    appear, allowing the user to decide if they can be safely
+    ignored.
     """
     distance_matrix = DistanceMatrix(distance_matrix)
 
-    E_matrix = e_matrix(distance_matrix.data)
-
-    # If the used distance was euclidean, pairwise distances
-    # needn't be computed from the data table Y because F_matrix =
-    # Y.dot(Y.T) (if Y has been centred).
-    F_matrix = f_matrix(E_matrix)
-
-    # If the eigendecomposition ever became a bottleneck, it could
-    # be replaced with an iterative version that computes the
-    # largest k eigenvectors.
-    eigvals, eigvecs = eigh(F_matrix)
-
-    # eigvals might not be ordered, so we order them (at least one
-    # is zero). cogent makes eigenvalues positive by taking the
+    # Center distance matrix, a requirement for PCoA here
+    matrix_data = center_distance_matrix(distance_matrix.data, inplace=inplace)
+
+    # If no dimension specified, by default will compute all eigenvectors
+    # and eigenvalues
+    if number_of_dimensions is None:
+        # distance_matrix is guaranteed to be square
+        number_of_dimensions = matrix_data.shape[0]
+    elif number_of_dimensions <= 0:
+        raise ValueError('Invalid operation: cannot reduce distance matrix '
+                         'to zero or negative dimensions using PCoA, '
+                         'must reduce to a minimum of 1.')
+
+    # Perform eigendecomposition
+    if method == "eigh":
+        # eigh does not natively support specifying number_of_dimensions, i.e.
+        # there are no speed gains unlike in FSVD. Later, we slice off unwanted
+        # dimensions to conform the result of eigh
+        # to the specified number_of_dimensions.
+
+        eigvals, eigvecs = eigh(matrix_data)
+        long_method_name = "Principal Coordinate Analysis"
+    elif method == "fsvd":
```

Comment:
Should we warn users if the number of dimensions is large when computing fsvd? If `number_of_dimensions` is `None` as is default, fsvd will compute a lot of dimensions which may be unexpected for the user user.

Reply:
Ok, I added a warning. "Large" is a bit hard to define, as any integer cutoff I add would feel a bit arbitrary, so I just added a general runtime warning of this behavior if the user invokes `fsvd` without specifying `number_of_dimensions`.

Reply:
I think the warning is in the wrong place? It looks like this warning will trigger even if the number of dimensions requested is 3?

Reply:
Change: the warning will now trigger of dimensions of matrix > 10.